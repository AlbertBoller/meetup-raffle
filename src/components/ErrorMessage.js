import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({
  title,
  titleClasses,
  subtitle,
  subtitleClasses,
  problemText,
  problemTextClasses,
  ...props
}) => (
  <section {...props}>
    <span className={titleClasses} data-testid="title">
      {title}
    </span>
    <p className={subtitleClasses} data-testid="subtitle">
      {subtitle}
    </p>
    <p className={problemTextClasses} data-testid="problemText">
      {problemText}
    </p>
  </section>
);

ErrorMessage.propTypes = {
  title: PropTypes.string,
  titleClasses: PropTypes.string,
  subtitle: PropTypes.string,
  subtitleClasses: PropTypes.string,
  problemText: PropTypes.string,
  problemTextClasses: PropTypes.string,
};

ErrorMessage.defaultProps = {
  title: 'Oops!',
  titleClasses: 'f4 b',
  subtitle: 'An error was encountered:',
  subtitleClasses: '',
  problemText: 'Unknown.',
  problemTextClasses: 'bg-red white courier pa2',
};

ErrorMessage.displayName = 'ErrorMessage';

export default ErrorMessage;
