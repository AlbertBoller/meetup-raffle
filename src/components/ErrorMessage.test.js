import React from 'react';
import { render } from 'react-testing-library';
import ErrorMessage from './ErrorMessage';

const errorProps = {
  title: 'Test error title',
  subtitle: 'Test error subtitle',
  problemText: 'Test error problem text',
};

describe('ErrorMessage', () => {
  const { container, queryByTestId } = render(<ErrorMessage />);

  it('renders', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders defaults', () => {
    render(<ErrorMessage />, { container });
    expect(queryByTestId('title')).toBeTruthy();
    expect(queryByTestId('subtitle')).toBeTruthy();
    expect(queryByTestId('problemText')).toBeTruthy();
  });

  it('renders overrides', () => {
    render(<ErrorMessage {...errorProps} />, { container });
    expect(queryByTestId('title').textContent).toBe(errorProps.title);
    expect(queryByTestId('subtitle').textContent).toBe(errorProps.subtitle);
    expect(queryByTestId('problemText').textContent).toBe(
      errorProps.problemText,
    );
  });
});
