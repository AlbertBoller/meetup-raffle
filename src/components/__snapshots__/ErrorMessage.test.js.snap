// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders 1`] = `
<div>
  <section>
    <span
      class="f4 b"
      data-testid="title"
    >
      Oops!
    </span>
    <p
      class=""
      data-testid="subtitle"
    >
      An error was encountered:
    </p>
    <p
      class="bg-red white courier pa2"
      data-testid="problemText"
    >
      Unknown.
    </p>
  </section>
</div>
`;
